src/components/implementation/no_interface/scheddev/sl_lock.c

```diff
@@ -0,0 +1,67 @@
+#include "sl.h"
+#include "sl_lock.h"
+
+void
+sl_lock_init(struct sl_lock *lock)
+{
+    assert(lock);
+    *lock = SL_LOCK_STATIC_INIT();
+}
+
+thdid_t
+sl_lock_holder(struct sl_lock *lock)
+{
+    assert(lock);
+    return lock->holder;
+}
+
+
+void
+sl_lock_lock(struct sl_lock *lock)
```

Comment:
It was hard to leave 80 comments.  It is hard to take the time to do that, and then have it take up *more* time the next time to send out a reminder.  Read the post on SWEng in research.  Even if code review comments are accidentally ignored (which is the case here), it is always hard for someone who puts in the time for the review to see it as anything other than you valuing your time other theirs.

Yes, code reviews suck for everyone.  But they are the only way to try and add some quality assurance and taste in the fundamental aspects of the system.  They're also a concrete way to do some training and to show more ways to think about code construction.  So it is important that they be taken seriously.

OK, I'm done ;-)